docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,236 @@
+- Any Fybrik operation that use the new change must not cause problems (e.g. crash or degrade behavior) when issued 
+against Fybrik objects, e.g. FybrikModules, Plotters, Blueprints and FybrikApplications created before the functionality was added.
+The objects should be smoothly promoted to the new version (if required).
+- It must be possible to round-trip Fybrik version changes (convert to different API versions and back) with no loss of 
+information (Should we support downgrade ?)
+
+*Note:* Fybrik allows to its users to [redefine most of the components](https://fybrik.io/v1.1/concepts/taxonomy/#taxonomy-contributors) 
+by using a custom taxonomy. If any of the components were redefined, it will be the user responsibility to update its 
+custom taxonomy files and use them during **each** 
+fybrik [upgrade](https://fybrik.io/v1.1/tasks/custom-taxonomy/#deploy-fybrik-with-custom-taxonomy). 
+
+## Fybrik Version Policy
+Fybrik uses [semantic versioning model](https://semver.org), which is the module adopted by a wide range of projects,
+e.g. Kubernetes, goLang, Helm and many others.
+
+Fybrik versions are of the form "vX.Y.Z" where X is the major version number, Y is the minor version number
+and Z is the patch release number.
+- Major Releases - released less frequently, indicate that **backward-incompatible public API changes**. This release
+  carries no guarantee that it will be backward compatible with preceding major versions.
+- Minor Releases - indicate **backward-compatible public API changes**. This release guarantees backward compatibility
+  and stability. The backward compatibility policy is defined in a separate document [ADD link] but in general, Fybrik
+  supports 2 previous minor releases.
+- Patch Releases - should not change main functionality, mostly for security or bug fixes, and therefore guarantee
+  backward compatibility.
+
+*Note:* Kubernetes uses similar semantic version model, but its versions do not have the "v" prefix. In order to use
+the same release tag and goLang module tag, we have chosen goLang releases modules pattern with the "v" prefix. The same
+version pattern will be used by all others, except CRDa, versioned Fybrik components. CRDs use its proprietary defined 
+by Kubernetes versioning.   
+
+### Backward compatibility policy
+Fybrik will support up to 2 backward minor releases. For example, the version v1.5.x should be able to work with 
+releases v1.4.x and v1.3.x, on another hand it might work with releases v1.2.x, but it is not guaranteed 
+
+## Fybrik components or features that can influence compatability 
+
+*Note:* the list is not closed, and can be extended
+
+*Note:* some components, listed below, are external integration points, changes in them impacts not only on the backward 
+compatibility, but on external development process. We emphasized these components with the label `external connector`.  
```

Comment:
Using the term "external connector" is confusing in my opinion since we have connectors that refer to something else.  I would suggest using the term integration point or something similar.